Adding Cray HPO tutorial content

ben-albrecht opened the pull request:
Adding Cray HPO SC19 tutorial content.

Commit:
hpo training basic example

Commit:
HPO content: sin, mnist_topo, hpo_train

ben-albrecht commented:
A few remaining TODOs:

- [ ] Update top-level `README.md`
- [ ] Include timing estimates in READMEs for users
- [x] Pull/rebase to resolve conflicts and use `--no-output` (thanks @sparticlesteve)

sparticlesteve commented on hpo_train.py:
```diff
@@ -0,0 +1,24 @@
+
+# Set up evaluator
+evaluator = hpo.Evaluator('python ./train.py configs/hpo_cifar10_cnn.yaml --hpo',
+                          nodes=4,
```
What does this arg do when the script is run within a slurm job? Is it just ignored?

sparticlesteve commented on train.py:
```diff
@@ -194,6 +195,8 @@ def main():
 
     if rank == 0:
+        if args.hpo:
+            logging.info('FoM: ' + str(history.history['val_loss'][0]))
```
Does this work? The logging will prepend the date and time.

Commit:
Improve examples

- Update default args
- Add some comments
- Infer number of nodes in batch scripts

Commit:
Revert scripts and configs to master

Commit:
Merge branch 'master' of github.com:NERSC/sc19-dl-tutorial into hpo-example
